core/app/models/spree/payment_method/store_credit.rb

```diff
@@ -84,8 +84,10 @@ def cancel(auth_code)
       if store_credit_event.nil? || store_credit.nil?
         ActiveMerchant::Billing::Response.new(false, '', {}, {})
       elsif store_credit_event.capture_action?
-        amount_in_cents = (store_credit_event.amount * 100).round
-        credit(amount_in_cents, auth_code)
+        credit(
+          store_credit_event.originator.credit_allowed_in_cents,
```

Comment:
I just assumed since a polymorphic association is used, we should expect it possible that something other than a payment method be the originator. Maybe that should be changed to not be polymorphic?